feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt

```diff
@@ -141,15 +148,19 @@ internal class HomeViewModel @Inject constructor(
             combine(
                 feedingPointStateFlow,
                 feedingStateFlow,
-                feedingRouteStateFlow
-            ) { feedingPointUpdate, feedingUpdate, feedingRouteUpdate ->
+                feedingRouteStateFlow,
+                stateFlow
+            ) { feedingPointUpdate, feedingUpdate, feedingRouteUpdate, homeStateFlow ->
                 updateState {
                     copy(
                         feedingPointState = feedingPointUpdate,
                         feedState = feedingUpdate,
                         feedingRouteState = feedingRouteUpdate,
                     )
                 }
+                if (feedingPointUpdate.feedingPoints.isNotEmpty() && !homeStateFlow.locationState.isUndefined) {
```

Comment:
Can you also refactor isUndefined logic within the current task? Remove underlying val's overriding